### PR TITLE
meta: remove legacy website code

### PR DESCRIPTION
**Pull request by ovflowd**
This PR removes the Legacy Website from the Codebase and automatically enables the Node.js Website Redesign (as the feature flag is unnecessary)

We can revert this anytime if we see issues or revert the Vercel deployment (without a Git revert).

Note that this is an initial cutover, and there might be more unused files, functions, and utilities that I might have missed, which can be cleaned up subsequently.

Also, legacy translations must be removed manually before we can enable Crowdin!

**THIS PR EFFECTIVELY ENABLES THE WEBSITE REDESIGN**

<sup>Do not merge this PR!</sup>

cc @nodejs/nodejs-website @nodejs/web-infra 

And I'm so freaking excited!

### CODE REVIEW

File: app/[locale]/layout.tsx [hunk 1]
```diff
@@ -4,29 +4,17 @@ import classNames from 'classnames';
...
 import { LocaleProvider } from '@/providers/localeProvider';
 import { ThemeProvider } from '@/providers/themeProvider';
 
-// Uses a WebPack/TurboPack Alias for resolving Global Styles
-// @deprecated remove when website redesign is done
-// eslint-disable-next-line import/no-unresolved
-import 'globalStyles';
+import '@/styles/index.css';
 
 // Defines the App Fonts based on being on Website Redesign or not
```
Comment (MattIPv4) [inline app/[locale]/layout.tsx]: This comment seems outdated now

File: .unimportedrc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  ],
+  "extensions": [".js", ".jsx", ".ts", ".tsx"],
+  "flow": false,
+  "ignorePatterns": [
+    "**/node_modules/**",
+    "**/*.tests.{js,jsx,ts,tsx}",
+    "**/*.test.{js,jsx,ts,tsx}",
+    "**/*.spec.{js,jsx,ts,tsx}",
+    "**/tests/**",
+    "**/__tests__/**",
```
Comment (AugustinMauroy) [inline .unimportedrc.json]:
```suggestion
    "**/__tests__/*.test.mjs",
```

in according with [`jest.conifg.mjs`](https://github.com/nodejs/nodejs.org/blob/main/jest.config.mjs)

Comment (ovflowd) [reply .unimportedrc.json]: Actually this file was commited accidentally hehe!

File: app/[locale]/layout.tsx (hunk 1 above)
Comment (ovflowd) [reply app/[locale]/layout.tsx]: I think I've removed the comment, no?

Comment (MattIPv4) [reply app/[locale]/layout.tsx]: no

File: next.constants.mjs [hunk 3]
```diff
@@ -113,7 +104,7 @@ export const MD_EXTENSION_REGEX = /((\/)?(index))?\.mdx?$/i;
  *
  * @todo: update the value when moving to website redesign
```
Comment (AugustinMauroy) [inline next.constants.mjs]: Is TODO still relevant?

**Review by AugustinMauroy — COMMENTED** [review]
I think we can remove all the `.hbs` in markown files (the key metadata `layout`) which are clearly legacy

**Comment by ovflowd** [comment]
> I think we can remove all the .hbs in markown files (the key metadata layout) which are clearly legacy

Not necessary for this PR, but we can do as a follow-up :)

**Review by MattIPv4 — APPROVED** [review]
YOLO

**Review by AugustinMauroy — APPROVED** [review]
LGTM !! Greeeattt we finally see what we've done 

Just here all i18n key can be remove (because you remove the component) _I checked by hand using VScode's search tool_

- `components.header.buttons.toggleLanguage`
- `components.home.x`
- `layouts.blogIndex.categoryName`
- `layouts.blogPost.author.byLine`

**Comment by ovflowd** [comment]
> LGTM !! Greeeattt we finally see what we've done
> 
> Just here all i18n key can be remove (because you remove the component) _I checked by hand using VScode's search tool_
> 
> * `components.header.buttons.toggleLanguage`
> * `components.home.x`
> * `layouts.blogIndex.categoryName`
> * `layouts.blogPost.author.byLine`

On a follow-up PR xD

**Review by bmuenzenmeyer — APPROVED** [review]

**Review by rxmarbles — APPROVED** [review]

**Review by shanpriyan — APPROVED** [review]

**Review by canerakdas — APPROVED** [review]
:shipit: ❤️

**Review by mikeesto — APPROVED** [review]
I've opened https://github.com/nodejs/nodejs.org/issues/6365 to address the breadcrumbs but let's get this merged

**Comment by ovflowd** [comment]
As per request from NVM maintainer, we removed the Windows option for "NVM" because it is not actually NVM, it's "NVM for Windows" which unfortunately uses the same name and creates confusion.

For Windows, we're recommending "Chocolatey". @nodejs/nodejs-website if y'all could re-review :)

**Comment by bmuenzenmeyer** [comment]
> For Windows, we're recommending "Chocolatey". @nodejs/nodejs-website if y'all could re-review :)

this looks fine, and i dont want any changes, but to be fair, I use "NVM for Windows"

**Comment by ovflowd** [comment]
> > For Windows, we're recommending "Chocolatey". @nodejs/nodejs-website if y'all could re-review :)
> 
> this looks fine, and i dont want any changes, but to be fair, I use "NVM for Windows"

Right, it's just that NVM for Windows is not NVM  and we want to avoid creating confusion.

**Review by canerakdas — APPROVED** [review]

**Comment by MattIPv4** [comment]
What's the plan for deploying this? Much as deploying on a Friday is generally a bad thing, part of me thinks it might be beneficial here to let this bake in production over the weekend with more hobbyist developer traffic, before the working week?

**Comment by ovflowd** [comment]
> What's the plan for deploying this? Much as deploying on a Friday is generally a bad thing, part of me thinks it might be beneficial here to let this bake in production over the weekend with more hobbyist developer traffic, before the working week?

I've made a https://beta-node-js-org.vercel.app/en and shared the link on Twitter. I believe we want to release this after a few feedbacks are in. We also want to write a blog post first (possibly) and get some things ready for launch.

**Comment by ovflowd** [comment]
cc @nodejs/nodejs-website @nodejs/web-infra 

I've made a new branch called "legacy" which is a stale copy of current latest commit of "main" and I've pointed Vercel to use "legacy" for production builds.

This allows us to merge this PR as it is now without accidentally releasing the website redesign on https://nodejs.org, and will enable us to keep the legacy website on maintenance mode (hot-fixes and bug-fixes on the "legacy" branch until we get the redesign released).

So, I'm merging this PR as it is, so we can adequately iterate feedback.

**Pull request merged**